source/kraken/apply_disruption.cpp

```diff
@@ -201,6 +195,149 @@ struct add_impacts_visitor : public apply_impacts_visitor {
+                route->for_each_vehicle_journey([&](nt::VehicleJourney& vj) {
+                    /*
+                     * Pre-filtering by validity pattern, which allows us to check if the vj is impacted quickly
+                     *
+                     * Since the validity pattern runs only by day not by hour, we'll compute in detail to
+                     * check if the vj is really impacted or not.
+                     *
+                     * */
+                    if ((vj.validity_patterns[rt_level]->days & impact_vp.days).none()) {
+                        return true;
+                    }
+
+                    // Filtering each journey to see if it's impacted by the section.
+                    SectionBounds bounds_st = vj.get_bounds_stop_times_for_section(ls.start_point, ls.end_point);
+                    // If the vj pass by both stops both elements will be different than nullptr, otherwise
+                    // it's not passing by both stops and should not be impacted
+                    if(bounds_st.first && bounds_st.second) {
+                        // Once we know the line section is part of the vj we compute the vp for the adapted_vj
+                        LOG4CPLUS_TRACE(log, "vj "<< vj.uri << " pass by both stops, might be affected.");
+                        nt::ValidityPattern new_vp{vj.validity_patterns[rt_level]->beginning_date};
+                        for(const auto& period : impact->application_periods) {
+                            // get the vp of the section
+                            new_vp.days |= vj.get_vp_for_section(bounds_st, rt_level, period).days;
+                        }
+                        // If there is effective days for the adapted vp we're keeping it
+                        if(!new_vp.days.none()){
+                            LOG4CPLUS_TRACE(log, "vj "<< vj.uri << " is affected, keeping it.");
+                            new_vp.days >>= vj.shift;
+                            has_affected_vj = true;
+                            vj_vp_pairs.emplace_back(&vj, new_vp, bounds_st);
+                        }
+                    }
+                    return true;
+                });
+
+                // The route isn't really impacted since no vj pass by the section
+                if(!has_affected_vj) {
+                    LOG4CPLUS_DEBUG(log, "Route  "<< route->uri << " has no affected vj, removing impact.");
+                    route->remove_impact(impact);
+                }
+            }
+
+            // Loop on each affected vj
+            for (auto& vj_vp_section : vj_vp_pairs) {
+                std::vector<nt::StopTime> new_stop_times;
+                const auto* vj = std::get<0>(vj_vp_section);
+                auto& new_vp = std::get<1>(vj_vp_section);
+                auto& bounds_st = std::get<2>(vj_vp_section);
+
+                bool ignore_stop(false);
+                for (const auto& st : vj->stop_time_list) {
```

Comment:
That's what I did first.
But I had a pair of uint_16 then, and I didn't really know what to set as a default value, std::numeric_limits<uint16_t>::max()? I need to have this default value to detect if one the stop isn't found in the vj. I switch to pointer for the easiness to compare to nullptr and direct access to the stop_time in the loop you commented on.

But I'm open to change it if you think it would be better to juste store the rank of the stop_times. I just need to know if you think storing std::numeric_limits<uint16_t>::max() as a default value is ok or if you suggest a different way to do it.